setup.py

```diff
@@ -103,11 +106,12 @@ def run(self):
     packages=find_packages("src"),
     package_dir={"": "src"},
     package_data={"imitation": ["py.typed", "envs/examples/airl_envs/assets/*.xml"]},
+    # Note: while we are strict with our test and doc requirement versions, we try to
+    #   impose as little restrictions on the install requirements as possible. Try to
+    #   encode only known incompatibilities here. This prevents nasty dependency issues
+    #   for our users.
     install_requires=[
-        # If you change gym version here, change it in "mujoco" below too.
-        # pinned to 0.21 until https://github.com/DLR-RM/stable-baselines3/pull/780
-        # goes upstream.
-        "gym[classic_control]==0.21.0",
+        "gym[classic_control]" + GYM_VERSION_SPECIFIER,
         "matplotlib",
         "numpy>=1.15",
         "torch>=1.4.0",
```

Comment:
should we also pin the packages listed in `PYTYPE = ["pytype"]` and in `extras_require["dev"]`?

Reply:
I agree that pinning pytype is good (although because they use a YYYY.MM.DD release format not semantic versioning it unfortunately may have to be an == not >=).

I think it's fine to leave the `"dev"` requires unpinned as not used in our CI.

In general I'd rather leave things unpinned except when needed for CI stability. Pinning forces us to manually update them which adds maintenance burden.

Reply:
True, I pinned it with `==` to the current version.